src/lib/cli.js

```diff
@@ -31,16 +31,34 @@ process.title = 'verdaccio';
 const logger = require('./logger');
 logger.setup(); // default setup
 
+const envinfo = require('envinfo');
 const commander = require('commander');
 const pkgVersion = module.exports.version;
 const pkgName = module.exports.name;
 
 commander
+  .option('--info', 'prints debugging information about the local environment')
   .option('-l, --listen <[host:]port>', 'host:port number to listen on (default: localhost:4873)')
   .option('-c, --config <config.yaml>', 'use this configuration file (default: ./config.yaml)')
   .version(pkgVersion)
   .parse(process.argv);
 
+if (commander.info) {
+	// eslint-disable-next-line no-console
+	console.log('\nEnvironment Info:');
+	envinfo
+	.run({
+	    System: ['OS', 'CPU'],
+	    Binaries: ['Node', 'Yarn', 'npm'],
```

Comment:
I was about to suggest `pnpm` 😢 but it does not show up in my local test, it is not detected unfortunately.